tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature

```diff
@@ -712,3 +712,39 @@ Feature: Sharing files and folders with internal users
       | shared-resource |
       | lorem.txt       |
       | simple-folder   |
+
+  Scenario: change existing expiration date of an existing share to a date beyond the enforced maximum expiry date
+    Given the setting "shareapi_default_expire_date_user_share" of app "core" has been set to "yes"
+    And the setting "shareapi_expire_after_n_days_user_share" of app "core" has been set to "7"
+    And user "user1" has created a new share with following settings
+      | path       | lorem.txt |
+      | shareWith  | user2     |
+      | expireDate | +7        |
+    And the setting "shareapi_enforce_expire_date_user_share" of app "core" has been set to "yes"
+    And user "user1" has logged in using the webUI
+    When the user tries to edit the collaborator expiry date of "User Two" of resource "lorem.txt" to "+14" days using the webUI
+    Then the expiration date field should be marked as "required" on the WebUI
+    And the expiration date for "+14 days" should be disabled
+    And the expiration date shown on the webUI should be "+7" days
+    And it should not be possible to save the pending share on the webUI
+
+  Scenario: setting default expiration date and enforcing it does not change the expiration date of a previously created share
+    Given user "user1" has created a new share with following settings
+      | path      | lorem.txt |
+      | shareWith | user2     |
+    And the setting "shareapi_default_expire_date_user_share" of app "core" has been set to "yes"
+    And the setting "shareapi_expire_after_n_days_user_share" of app "core" has been set to "7"
+    And the setting "shareapi_enforce_expire_date_user_share" of app "core" has been set to "yes"
+    When user "user1" logs in using the webUI
```

Comment:
@jasson99 @individual-it I think according to the scenario, it should be like:
```suggestion
    When the administrator sets the setting "shareapi_enforce_expire_date_user_share" of app "core" to "yes"
    And user "user1" logs in using the webUI
```

Reply:
@kiranparajuli589 , I think we can go both ways for this. I still think the step can be given step. Moreover, we need to write extra step definition in case we use this as when. So for now i would like to leave it as it is